src/libraries/System.Console/src/System/Console.cs

```diff
@@ -76,11 +72,7 @@ public static Encoding InputEncoding
                 {
                     lock (s_syncObject)
                     {
-                        if (s_inputEncoding == null)
-                        {
-                            Volatile.Write(ref s_inputEncoding, ConsolePal.InputEncoding);
-                        }
-                        encoding = s_inputEncoding;
+                        encoding = s_inputEncoding ??= ConsolePal.InputEncoding;
```

Comment:
My understanding is these volatile writes are here to ensure the store happens after the object is fully initialized.

Otherwise:
```
Encoding? encoding = Volatile.Read(ref s_inputEncoding);
```
may read an encoding that is still being initialized on another thread.

Reply:
fyi, we had this discussion once about removing the volatile writes in this code: https://github.com/dotnet/runtime/pull/34297#discussion_r404804789.